test/node-test.js

```diff
@@ -29,43 +29,35 @@ describe('node', function () {
     });
   });
 
-  it('fails', function (done) {
+  it('fails and continues if `it` throws', function (done) {
     run('fails', ['--node', '-R', 'tap'], function (code, stdout) {
-      assert.equal(stdout.indexOf('# node:\n'
-        + '1..1\n'
-        + 'not ok 1 test fails\n'
-        + '  Error: Oh noes!'), 0);
       assert.equal(code, 1);
+
+      var lines = stdout.trim().split(/\n+/);
+      assert.equal(lines[0], '# node:');
+      assert.equal(lines[1], '1..4');
+      assert.equal(lines[2], 'not ok 1 test fails synchronously');
+      assert.equal(lines[3], '  Error: Oh noes!');
+      assert.equal(lines[5], 'not ok 2 test fails asynchronously');
+      assert.equal(lines[6], '  Error: Oh noes!');
+      assert.equal(lines[8], 'ok 3 test passes synchronously');
+      assert.equal(lines[9], 'ok 4 test passes asynchronously');
+      assert.equal(lines[10], '# tests 4');
+      assert.equal(lines[11], '# pass 2');
+      assert.equal(lines[12], '# fail 2');
       done();
     });
   });
 
-  it('handles async failures', function (done) {
-    run('fails-async', ['--node', '-R', 'tap'], function (code, stdout) {
+  it('fails and exits if `describe` throws', function (done) {
+    run('describe-throws', ['--node', '-R', 'tap'], function (code, stdout) {
+      assert.equal(stdout.indexOf('# node:'), 0);
```

Comment:
Somehow this is all that node prints in this situation, I have to admit I don't know who is eating the error message here?